### PR TITLE
Change kube-proxy & fluentd CPU request to 20m/80m.

### PR DESCRIPTION
**Pull request by cjcullen**
On 1.1.8, our per-node CPU overhead was 100m (fluentd 100m, kube-proxy unaccounted). To stay at that number, we should adjust fluentd to 80m, and give kube-proxy 20m.

Fixes the "failure to schedule kube-dns" part of #23556.

### CODE REVIEW

**Comment by bprashanth** [comment]
I think this is a good compromise for right now. Unfortunately kube-proxy will still get starved when all the addons land on the same node, becaues they all have higher limits, which leads to a bad ux since services are a core part of our system, but this is cpu so it should be OK.

We will have to bump up these limits sometime. I'm not sure how we'd do that yet. 

This might also result in test flake, but we can handle that as it comes, we can easily figure out a higher timeout till a roll-forward solution presents itself.

**Comment by cjcullen** [comment]
I'm going to switch it to 75/25 for fluentd/kube-proxy. In 1.1.8, kube-proxy had no reservation, so this will still be strictly better for kube-proxy, while not constraining fluentd too much.

File: cluster/saltbase/salt/kube-proxy/init.sls [hunk 1]
```diff
@@ -17,7 +17,7 @@
     - makedirs: true
     - dir_mode: 755
     - context:
-        cpurequest: '200m'
+        cpurequest: '20m'
```
Comment (bprashanth) [inline cluster/saltbase/salt/kube-proxy/init.sls]: please add a comment with something like:
```
// 20m might cause cpu starvation for kube-proxy resulting in delayed service updates. However 
// we can't given it more cpu till we have a way to do so without affecting existing Kubernetes clusters. 
// Any change made here must reflect in a proportional change to the requests of other daemons on
// the node, eg fluentd.
```

Comment (bprashanth) [inline cluster/saltbase/salt/kube-proxy/init.sls]: I meant # wherever I said //

**Comment by alex-mohr** [comment]
And documenting for the record, @a-robinson mentioned that fluentd in 1.2 is more efficient than 1.1, so we suspect there won't be much if any regression in user-visible logging performance by downsizing it from 100m to 80m.

File: cluster/saltbase/salt/kube-proxy/init.sls (hunk 1 above)
Comment (cjcullen) [inline cluster/saltbase/salt/kube-proxy/init.sls]: How's that look?

File: cluster/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ spec:
       limits:
         memory: 200Mi
       requests:
-        cpu: 100m
+        cpu: 80m
```
Comment (bprashanth) [inline cluster/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml]: please add the same note of caution here.

File: cluster/saltbase/salt/kube-proxy/init.sls [hunk 3]
```diff
@@ -17,7 +17,12 @@
     - makedirs: true
     - dir_mode: 755
     - context:
-        cpurequest: '200m'
+    # 20m might cause kube-proxy CPU starvation on full nodes, resulting in
```
Comment (bprashanth) [inline cluster/saltbase/salt/kube-proxy/init.sls]: nit: please indent with cpurequest (I wouldn't bring this nit up if I didn't already have a reason to re-push, which is the other comment)

**Comment by bprashanth** [comment]
Your bug is https://github.com/kubernetes/kubernetes/issues/23659 in case you really need to push this out and can't re-up

File: cluster/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml (hunk 2 above)
Comment (cjcullen) [inline cluster/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml]: Done.

File: cluster/saltbase/salt/kube-proxy/init.sls (hunk 3 above)
Comment (cjcullen) [inline cluster/saltbase/salt/kube-proxy/init.sls]: Done.

File: cluster/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml [hunk 4]
```diff
@@ -14,7 +14,9 @@ spec:
       limits:
         memory: 200Mi
       requests:
-        cpu: 100m
+	    # Any change here should be accompanied by a proportional change in CPU
+	    # requests of other per-node add-ons (e.g. kube-proxy).
```
Comment (bprashanth) [inline cluster/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml]: tabs vs white space? 
or is github just nuts

Comment (cjcullen) [inline cluster/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml]: Uggh. Shouldn't have edited these in eclipse...

**Comment by bprashanth** [comment]
Thanks

**Comment by cjcullen** [comment]
Looks like 20m may not be enough to avoid flakes in kube-proxy: 
```
Failure [270.683 seconds]
[k8s.io] Services
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework.go:420
  should be able to up and down services [It]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/service.go:276

  Expected error:
      <*errors.errorString | 0xc20904ddb0>: {
          s: "Only 2 pods started out of 3",
      }
      Only 2 pods started out of 3
  not to have occurred
```

**Comment by cjcullen** [comment]
Although that doesn't sound like a kube-proxy flake...

**Comment by cjcullen** [comment]
@k8s-bot unit test this github issue: #23668

**Comment by cjcullen** [comment]
@k8s-bot test this github issue: #23668

**Comment by alex-mohr** [comment]
Mergebot is blocked on unrelated kubemark test suite broken for past 5 hours.  This PR passed tests, is low risk, and fixes a significant regression in the amount of cpu available to users on each node, so manually merging.

**Comment by wojtek-t** [comment]
@gmarek - FYI

**Comment by david-mcmahon** [comment]
The cherrypick process is defined here.  Hoping to get this PR merged today, but the details are [here](https://github.com/kubernetes/kubernetes/pull/23555/files).  You need to run the cherry pick tool yourself to get these onto the branch.  @bgrant0607 will then approve them for merge.

**Comment by bgrant0607** [comment]
Every cherrypick needs a release note.

**Comment by bgrant0607** [comment]
This PR has no description and no reference to a P0 issue.

**Comment by cjcullen** [comment]
Fixes the "failure to schedule kube-dns" part of #23556.

**Comment by cjcullen** [comment]
Internal bugref: 27924350

**Comment by fagiani** [comment]
@cjcullen thanks for this PR, it helps a lot to get smaller clusters running smooth. I've tested with 1 master and 2 node hosts on AWS m3.medium machines and it works great!

**Comment by thockin** [comment]
@vishh @dchen1107 how far are we from having a way to use docker's --cgroup-parent logic?  This might make a cool intern project if we can spec it enough.  Then we could have a handful of special cgroups (let's call it "sys" for things that need to run on each machine) that some pods can ask to be contained in.  I doubt that is the right API but we can work on that as we spec...

**Comment by vishh** [comment]
@thockin that's the plan. My intern is most likely working on cgroups.

**Comment by vishh** [comment]
BTW we already use cgroup parent, except that it is not configurable per
pod.

**Comment by davidopp** [comment]
I don't think that will solve the problem of occasionally needing to increase the sys cgroup's size from time to time, though.

**Pull request merged**